JumpscaleCore/tools/threebot/ThreebotExplorer.py

```diff
@@ -161,3 +167,98 @@ def sign(nacl, *args):
         self._log_info("registration of threebot '{%s}' done" % name)
 
         return record1
+
+    def _reservation_create(self, flist="", hub_url="", environment={}, entrypoint=""):
+        """
+        Create a reservation object with container workload from schema url of "tfgrid.reservation.1".
+        The parameters passed will be added to the container workload
+
+        :param flist: link to flist to create a container from
+        :type flist: str
+        :param hub_url: link to hub
+        :type hub_url: str
+        :param environment: environment variables to be passed to the contiainer to be created eg : {"USER":"test_user"}
+        :type environment: dict
+        :param entrypoint: entrypoint to container start
+        :type entrypoint: str
+        :return: reservation object
+        :return type: obj
+        """
+        reservation_model = j.data.schema.get_from_url(url="tfgrid.reservation.1")
+
+        reservation = reservation_model.new()
+        reservation.customer_tid = j.tools.threebot.me.default.tid
+        reservation.data_reservation.expiration_provisioning = int(j.data.time.epoch + 30 * 60)
```

Comment:
These 2 expiration value need to be configurable. Having default value for them is ok, but not just hardcoded.

Reply:
Yes I'll  make them configurable and add default value if they were not given.